File: 프로그래머스/2/42586. 기능개발/기능개발.py
from math import ceil

def solution(progresses, speeds):
    result = []
    count = 0
    days = [ceil((100 - p) / s) for p, s in zip(progresses, speeds)]
    
    print("first days :", days)
    while True:
        if count == 0:
            days = [day - 1 for day in days]
        
        tmp = len(days)
        
        for i in range(len(days)):
            if i == 0 and days[i] <= 0:
                days.pop(0)
                count += 1
                break
        
        if len(days) == tmp and count > 0:
            result.append(count)
            count = 0
            if len(days) == 0:
                break
        else:
            pass
    return result

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <stack>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    stack<int> s;
    int length = progresses.size();
    int day = 0;
    int cnt;
    
    for (int i = length-1; i > -1; i--) {
        s.push((100-progresses[i] + speeds[i] - 1) / speeds[i]);
    }
    
    while (!s.empty()) {
        if (s.top() == day) {
            cnt = 0;
            while (!s.empty() and s.top() <= day) {
                s.pop();
                cnt++;
            }
            answer.push_back(cnt);
        }
        day++;
    }
    return answer;
}